tests/testLoginManager.php

```diff
@@ -133,4 +173,131 @@ function ( $auth_url, $params ) {
 		);
 		$this->assertEquals( 'https://test-custom.auth0.com/authorize?test=this', $auth_url );
 	}
+
+	/**
+	 * Test that logged-in users are redirected from the wp-login.php page.
+	 */
+	public function testThatLoggedInUserIsRedirectedFromWpLogin() {
+		$this->startRedirectHalting();
+
+		$login_manager = new WP_Auth0_LoginManager( self::$users_repo, self::$opts );
+
+		// Configure Auth0.
+		self::auth0Ready();
+		$this->assertTrue( WP_Auth0::ready() );
+
+		// Set the current user to admin.
+		$this->setGlobalUser();
+
+		// Use the default login redirection.
+		$caught_redirect = [
+			'location' => null,
+			'status'   => null,
+		];
+		try {
+			$login_manager->login_auto();
+		} catch ( Exception $e ) {
+			$caught_redirect = unserialize( $e->getMessage() );
+		}
+
+		// Redirect will have a dynamic cache breaker so cannot assertEquals here.
+		$this->assertNotFalse( strpos( $caught_redirect['location'], 'http://example.org?' ) );
+		$this->assertEquals( 302, $caught_redirect['status'] );
+
+		// Set a one-time login redirect URL.
+		$_REQUEST['redirect_to'] = 'http://example.org/custom';
+
+		// Use the default login redirection.
+		$caught_redirect = [
+			'location' => null,
+			'status'   => null,
+		];
+		try {
+			$login_manager->login_auto();
+		} catch ( Exception $e ) {
+			$caught_redirect = unserialize( $e->getMessage() );
+		}
+
+		// Redirect will have a dynamic cache breaker so cannot assertEquals here.
+		$this->assertNotFalse( strpos( $caught_redirect['location'], 'http://example.org/custom?' ) );
+		$this->assertEquals( 302, $caught_redirect['status'] );
+	}
+
+	/**
+	 * Test that the ULP redirect happens.
+	 */
+	public function testUlpRedirect() {
```

Comment:
Set everything up to make sure the ULP redirect happens under normal conditions.